### PR TITLE
Websocket race

### PR DESCRIPTION
**Pull request by vtereso**
This PR is a fix for issue https://github.com/tektoncd/dashboard/issues/73
---
Slight updates to broadcaster and websocket to prevent races
Added _testing.go files for broadcaster (websocket is tested from endpoints)

### CODE REVIEW

**Comment by a-roberts** [comment]
Needs squashing first please

**Comment by a-roberts** [comment]
`I0426 19:27:06.505] error while parsing /go/src/github.com/tektoncd/dashboard/Gopkg.lock: Unable to parse the lock as TOML: (615, 1): no value can start with <`

Incomplete merge, see [here](https://github.com/tektoncd/dashboard/pull/94/files#diff-bd247e83efc3c45ae9e8c47233249f18R615), so close!

File: pkg/broadcaster/broadcaster_test.go [hunk 1]
```diff
@@ -0,0 +1,166 @@
+package broadcaster
+
+import (
+	"sync"
+	"testing"
+)
+
+// Add and remove Subscribers
```
Comment (vdemeester) [inline pkg/broadcaster/broadcaster_test.go]: This file needs to be passed through `gofmt` (we need to enable a check on that :angel: )

File: Gopkg.lock [hunk 2]
```diff
@@ -612,6 +612,14 @@
     "github.com/tektoncd/pipeline/pkg/client/clientset/versioned",
     "github.com/tektoncd/pipeline/pkg/client/clientset/versioned/fake",
     "github.com/tektoncd/pipeline/pkg/client/informers/externalversions",
+<<<<<<< HEAD
```
Comment (vdemeester) [inline Gopkg.lock]: The conflict is here :sweat_smile: Remove the conflict marker and re-run `dep ensure`, it should fix itself :wink:

File: pkg/websocket/websocket.go [hunk 3]
```diff
@@ -42,75 +42,77 @@ func WriteOnlyWebsocket(connection *websocket.Conn, b *broadcaster.Broadcaster)
...
+		// Ellicit another ping
+		writePing(connection, deadline)
 		return nil
 	})
-	time.Sleep(time.Second)
+	initialDeadline := time.Now().Add(time.Second)
+	connection.SetReadDeadline(initialDeadline)
+	// Kick off cycle
+	writePing(connection, initialDeadline)
 	for {
```
[the remark this answers is not in the document]
Comment (vtereso) [reply pkg/websocket/websocket.go]: ReadMessage is blocking until it receives a text message while control messages received within this call are passed off to the respective handlers. The connection is constantly being read to detect failures from the client (read timeout if pongs are received or normal closure).  We don't expect any `text` messages from the client as we are treating the websocket as a server side event stream, not a duplex connection stream. So, assuming any message is received, we just enter read message again.

[the remark this answers is not in the document]
Comment (ncskier) [reply pkg/websocket/websocket.go]: Ok, sounds good. I just wanted to verify that it is blocking, so we don't have this `for` loop hogging computing power when it doesn't need to be.

**Comment by a-roberts** [comment]
Thanks for reviewing this @ncskier, can you give us an update on this one please @vtereso? I'd like us to get the -race option added in as soon as we can before we land more goodies

**Pull request merged**